server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/BulkApplyTemplateAction.java

```diff
@@ -91,6 +104,40 @@ public void define(WebService.NewController context) {
       .setDeprecatedKey(PARAM_QUALIFIER, "6.6");
 
     createTemplateParameters(action);
+
+    action
+      .createParam(PARAM_PROJECTS)
+      .setDescription("Comma-separated list of project keys")
+      .setSince("6.6")
+      .setExampleValue(String.join(",", KEY_PROJECT_EXAMPLE_001, KEY_PROJECT_EXAMPLE_002));
+
+    action
+      .createParam(PARAM_PROJECT_IDS)
+      .setDescription("Comma-separated list of project ids")
+      // to map parameters of api/projects/search
+      .setSince("6.6")
+      .setDeprecatedSince("6.6")
+      .setExampleValue(String.join(",", UUID_EXAMPLE_01, UUID_EXAMPLE_02));
+
+    action.createParam(PARAM_VISIBILITY)
```

Comment:
Is it correct that this parameter is not used ?

Reply:
If it's not used, I wouldn't add it

Reply:
I haven't checked yet, but to be consistent either it's available in api/projects/search and this WS or not at all